homeassistant/components/melnor/switch.py

```diff
@@ -48,12 +48,11 @@ def __init__(
         super().__init__(coordinator)
         self._valve_index = valve_index
 
+        self._attr_name = f"Zone {self._valve().id+1}"
         self._attr_unique_id = (
-            f"switch-{self._attr_unique_id}-zone{self._valve().id}-manual"
+            f"switch-{self._device.mac}-zone{self._valve().id}-manual"
```

Comment:
If you use an intermediate variable to get the valve id, you can reduce this to three lines instead of four.
```python
        valve_id = self._valve().id
        self._attr_name = f"Zone {valve_id+1}"
        self._attr_unique_id = f"switch-{self._device.mac}-zone{valve_id}-manual"
```

Also, slightly unrelated but `_valve_index: int` on line 39 is not needed as it is inferred from the definition in `__init__`

Reply:
Made the first change. I'll address `_valve_index` in another PR (because you're right) as I rework the switch platform to use `entity_description` and add more switches.